share/availability/discovery/discovery.go

```diff
@@ -242,6 +242,10 @@ func (d *Discovery) ensurePeers(ctx context.Context) {
 	defer t.Stop()
 	for {
 		d.findPeers(ctx)
+		if d.set.Size() < d.set.Limit() {
+			// rerun discovery is amount of peers didn't reach the limit
+			continue
+		}
```

Comment:
From discussion with Ryan I get the impression that you decided against that

Reply:
Previous implementation had a lot of issues. This one is doing good, because of other changes to errorgroup. Also in reality, when discovery is launched it is only able to find local host for few seconds, and this check allows it to find legit peers as soon as possible.